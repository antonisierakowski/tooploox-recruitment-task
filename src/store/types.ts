import { GhUserState } from './ghUser/types';
import { NotificationsState } from './notifications/types';

export interface RootState {
  // ghUser: GhUserState;
  // notifications: NotificationsState
}

export interface Action<TPayload = any> {
  type: string;
  payload?: TPayload;
}
