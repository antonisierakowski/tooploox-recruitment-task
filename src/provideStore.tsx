import React from 'react';
import { Store } from 'redux';
import { Provider } from 'react-redux';

export const provideStore = (
  component: React.ReactElement,
  store: Store,
): React.ReactElement => <Provider store={store} children={component} />;
