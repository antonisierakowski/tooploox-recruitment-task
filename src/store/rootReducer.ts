import { combineReducers } from 'redux';
import { notificationsReducer } from './notifications/reducer';
import { ghUserReducer } from './ghUser/reducer';

export const rootReducer = combineReducers({
  ghUserState: ghUserReducer,
  notificationsState: notificationsReducer,
});
