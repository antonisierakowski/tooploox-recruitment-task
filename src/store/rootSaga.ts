import { all } from 'redux-saga/effects';
import { notificationSaga } from './notifications/sagas';

export function* rootSaga() {
  yield all([notificationSaga()]);
}
