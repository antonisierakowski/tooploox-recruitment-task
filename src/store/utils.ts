import { Action } from './types';

export function createAction<TPayload>(
  actionType: string,
  payload?: TPayload,
): Action<TPayload> {
  return {
    type: actionType,
    payload,
  };
}
