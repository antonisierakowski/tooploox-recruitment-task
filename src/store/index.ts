import { createStore, applyMiddleware, Store } from 'redux';
import { rootReducer } from './rootReducer';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension';
import { Action, RootState } from './types';
import { rootSaga } from './rootSaga';

const sagaMiddleware = createSagaMiddleware();

const middleware = composeWithDevTools(applyMiddleware(sagaMiddleware));

export const store: Store<RootState, Action> = createStore(
  rootReducer,
  middleware,
);

export const createPreloadedStore = (
  state: RootState,
): Store<RootState, Action> => createStore(rootReducer, state, middleware);

sagaMiddleware.run(rootSaga);
