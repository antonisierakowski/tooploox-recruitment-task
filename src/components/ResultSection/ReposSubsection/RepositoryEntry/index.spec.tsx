import { render } from '@testing-library/react';
import { provideStore } from '../../../../provideStore';
import React from 'react';
import { RepositoryEntry } from './index';
import { createPreloadedStore } from '../../../../store';
import { rootStateFixture } from '../../../../fixtures/rootStateFixture';

const store = createPreloadedStore(rootStateFixture);

it('should match to the generated snapshot', () => {
  const { asFragment } = render(
    provideStore(<RepositoryEntry id={1} />, store),
  );
  expect(asFragment()).toMatchSnapshot();
});
