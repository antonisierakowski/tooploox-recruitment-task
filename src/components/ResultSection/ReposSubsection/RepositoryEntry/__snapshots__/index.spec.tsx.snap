// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match to the generated snapshot 1`] = `
<DocumentFragment>
  <div
    class="repository"
  >
    <a
      href="test"
    >
      test1
    </a>
    <span
      class="starsIndicator"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      1
    </span>
  </div>
</DocumentFragment>
`;
